public/assets/js/module/Modal.mjs

```diff
@@ -0,0 +1,372 @@
+			element.classList.add("img");
+			element.style.setProperty("background-position-y",`calc(var(--sprite-thumbnail-height) * -${index})`);
+
+			return element;
+		}
+
+		let item = Modal.createDiv("item center");
+		item.setAttribute("data",data);
+
+		// Treat item as multipack if index is false
+		if(index === false) {
+			item.appendChild(Modal.textElement(text));
+
+			return item;
+		}
+
+		item.appendChild(spriteElement(index));
+		item.appendChild(Modal.textElement(text));
+		item.appendChild(Summary.inputElement(data));
+
+		return item;
+	}
+
+	appendProducts(selected) {
+		const list = Modal.createDiv("list");
+
+		const bulk = Modal.createDiv("bulk center");
+		bulk.appendChild(Modal.textElement("Buy your configuration in bulk:"));
+		bulk.appendChild(Summary.inputElement("bulk"));
+
+		this.append(bulk);
+
+		for(const [id,index] of Object.entries(selected)) {
+			let data = this.products[id];
+			let value = data.add.length > 0 ? data.add[0] : id;
+
+			let item = this.createListItem(index,id,value);
+			list.appendChild(item);
+			
+			// Add multipack products if array length is > 0
+			if(data.add.length > 0) {
+				data.add.slice(1).forEach(pointer => {
+					const data = this.products[pointer];
+					const value = data.add.length > 0 ? data.add[0] : pointer;
+
+					let item = this.createListItem(false,pointer,value);
+					item.setAttribute("multipack",id);
+
+					list.appendChild(item);
+				});
+			}
+		}
+
+		this.append(list);
+	}
+
+	appendCheckout() {
+		const button = Modal.createDiv("button");
+		button.setAttribute("data","addtocart");
+		button.innerText = "Add to Cart";
+
+		this.append(button);
+	}
+
+}
+
+export class Categories extends Modal {
+
+	constructor(categories) {
+		super("Categories");
+		this.inner.classList.add("categories");
+
+		this.appendCategories(categories);
+		this.appendSummary();
+		this.bind();
+
+		this.open();
+	}
+
+	click(event) {
+		const target = event.target.closest(".item") ?? false;
+		if(!target) {
+			return false;
+		}
+
+		let value = parseInt(target.getAttribute("data"));
+
+		if(value === 0) {
+			value = "summary";
+		}
+
+		window._configCategory(value);
+		this.close();
+	}
+
+	bind() {
+		this.inner.addEventListener("click",event => this.click(event));
+	}
+
+	// ----
+
+	createListItem(icon,text,id) {
+		let item = Modal.createDiv("item center");
+		item.setAttribute("data",id);
+
+		item.appendChild(Modal.textElement(icon));
+		item.appendChild(Modal.textElement(text));
+
+		return item;
+	}
+
+	appendCategories(categories) {
+		let list = Modal.createDiv("list");
+
+		let i = 1;
+		for(let [id,data] of Object.entries(categories)) {
+			if(!data.enabled) {
+				continue;
+			}
+
+			list.appendChild(this.createListItem(i,data.name,id));
+			i++;
+		}
+
+		this.append(list);
+	}
+
+	appendSummary() {
+		const list = this.inner.getElementsByClassName("list")[0];
+
+		list.appendChild(this.createListItem("i","Summary",0));
+	}
+
+}
```

Comment:
Public class field not supported in Safari 14.

[MDN Reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields)
[WebKit bug report #194095](https://bugs.webkit.org/show_bug.cgi?id=194095)

Reply:
Fixed in ba7800070cda626a6c0873ce7e3afdc456a138ca